lib/internal/streams/from.js

```diff
@@ -53,18 +50,22 @@ function from(Readable, iterable, opts) {
   };
 
   readable._destroy = function(error, cb) {
-    if (needToClose) {
-      needToClose = false;
-      close().then(
-        () => process.nextTick(cb, error),
-        (e) => process.nextTick(cb, error || e),
-      );
-    } else {
-      cb(error);
-    }
+    close(error).then(
+      () =>  cb(error),
+      (e) => cb(error ?? e)
+    )
   };
 
-  async function close() {
+  async function close(error) {
```

Comment:
This is tricky and I want feedback on the semantics, basically:
 - If we had an error ( `.destroy(err)` ) we call `.throw` - this aligns the behaviour of `.from` with other methods and means stuff like `addAbortSignal` automagically "works".
 - If after calling `throw` the iterator did not finish - we also call `.return`